services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/FlowService.java

```diff
@@ -102,10 +107,16 @@ public FlowPair createFlow(Flow flow, FlowCommandSender sender) throws Recoverab
         FlowPairWithSegments result = transactionManager.doInTransaction(() -> {
             FlowPair flowPair = flowResourcesManager.allocateFlow(buildFlowPair(flow, pathPair));
 
-            flowRepository.createOrUpdate(flowPair);
+            List<FlowSegment> forwardSegments = buildFlowSegments(flowPair.getForward());
+            List<FlowSegment> reverseSegments = buildFlowSegments(flowPair.getReverse());
+            List<FlowSegment> flowSegments = Stream.of(forwardSegments, reverseSegments)
```

Comment:
Nice java-8-ish style, but union(forwardSegments, reverseSegments) is shorter :) with static import of CollectionUtils.union

Reply:
Fixed.